### PR TITLE
Allow non-array json string

### PR DESCRIPTION
**Pull request by sw7190**
Allow non-array json string to ParseJSONRuleset

### CODE REVIEW

**Comment by niallnsec** [comment]
Hi sw7190,

The ParseJSONRuleset function is explicitly meant to parse a JSON ruleset into GRL syntax and as such expects the input to be an array of JSON objects. I think parsing a single JSON rule into GRL syntax would call for a separate function to be defined called ParseJSONRule which would do what you are after. Beyond that I think it makes sense to have a ParseJSONRule function anyway since it would be useful to anyone storing GRL/JSON rules in a database table or similar storage.

I am assuming you want to be able to load a single rule using the JSONResource loader, is that correct? If that is the case, I think a cleaner approach is to define a new function as described above and then in JSONResource.Load we can switch on the first rune of the input data. If it is a '[' we would call ParseJSONRuleset and if it is. a '{' we would call ParseJSONRule.

**Comment by sw7190** [comment]
> Hi sw7190,
> 
> The ParseJSONRuleset function is explicitly meant to parse a JSON ruleset into GRL syntax and as such expects the input to be an array of JSON objects. I think parsing a single JSON rule into GRL syntax would call for a separate function to be defined called ParseJSONRule which would do what you are after. Beyond that I think it makes sense to have a ParseJSONRule function anyway since it would be useful to anyone storing GRL/JSON rules in a database table or similar storage.
> 
> I am assuming you want to be able to load a single rule using the JSONResource loader, is that correct? If that is the case, I think a cleaner approach is to define a new function as described above and then in JSONResource.Load we can switch on the first rune of the input data. If it is a '[' we would call ParseJSONRuleset and if it is. a '{' we would call ParseJSONRule.

Okay. sure!

File: pkg/JsonResource.go [hunk 1]
```diff
@@ -56,7 +57,15 @@ func (jr *JSONResource) Load() ([]byte, error) {
...
-	rs, err := ParseJSONRuleset(data)
+	firstRune := string(bytes.TrimSpace(data)[0])
+
+	var rs string
+
+	if firstRune == "[" {
+		rs, err = ParseJSONRuleset(data)
+	} else if firstRune == "{" {
+		rs, err = ParseJSONRule(data)
+	}
```
Comment (niallnsec) [inline pkg/JsonResource.go:68]:
```suggestion
	} else {
		err = errors.New("invalid JSON input")
	}
```
Looks good! We just need to catch the extra case where we may have loaded something which is not valid JSON.

**Review by niallnsec — CHANGES_REQUESTED** [review]
This all looks good with the one small change needed to catch any invalid JSON being passed in to the JSONResource Load function.

**Comment by sw7190** [comment]
> JSONResource Load 함수로 전달되는 유효하지 않은 JSON을 잡는 데 필요한 작은 변경 하나만으로 이 모든 것이 좋아 보입니다.

Good!

File: pkg/JsonResource.go [hunk 2]
```diff
@@ -15,6 +15,7 @@
 package pkg
 
 import (
+	"bytes"
 	"encoding/json"
```
Comment (niallnsec) [inline pkg/JsonResource.go:19]:
```suggestion
	"encoding/json"
	"errors"
```

**Review by niallnsec — CHANGES_REQUESTED** [review]
Need to fix missing import causing build failure

**Review by niallnsec — APPROVED** [review]

**Comment by newm4n** [comment]
@sw7190 
Seems your branch is out-of-date
Could your update your branch to the latest upstream master.
I will merge this as soon as its ready.

**Pull request merged**